pkg/apis/kops/validation/legacy.go

```diff
@@ -289,7 +288,7 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 
 			// @ check that NodeLocalDNS addon is configured correctly
 			if c.Spec.KubeDNS.NodeLocalDNS != nil && fi.BoolValue(c.Spec.KubeDNS.NodeLocalDNS.Enabled) {
-				if c.Spec.KubeDNS.Provider != "CoreDNS" && (c.Spec.KubeDNS.Provider != "" || c.IsKubernetesLT("1.20")) {
+				if c.Spec.KubeDNS.Provider != "CoreDNS" && c.Spec.KubeDNS.Provider != "" {
```

Comment:
Does this mean we can remove the KubeDNS addon?

Reply:
This test is only for dns cache, I think.